front-end/src/components/BlockCountdown.tsx

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { bnToBn } from '@polkadot/util';
+import BN from 'bn.js';
+import React, { useContext, useEffect, useState } from 'react';
+import { Popup } from 'semantic-ui-react';
+import { ApiContext } from 'src/context/ApiContext';
+import { chainProperties } from 'src/global/networkConstants';
+import blockToTime from 'src/util/blockToTime';
+import getNetwork from 'src/util/getNetwork';
+
+interface Props {
+	className?: string
+	endBlock: number;
+}
+
+const BlockCountdown = ({ className, endBlock }:Props ) => {
+	const network = getNetwork();
+	const ZERO = new BN(0);
+	const { api, apiReady } = useContext(ApiContext);
+	const [currentBlock, setCurrentBlock] = useState(ZERO);
+	const blocksRemaining = endBlock - currentBlock.toNumber();
+	const DEFAULT_TIME = bnToBn(chainProperties?.[network]?.blockTime);
+	const [blocktime, setBlocktime] = useState(DEFAULT_TIME);
+
+	useEffect(() => {
+		if (!api) {
+			console.error('polkadot/api not set');
+			return;
+		}
+
+		if (!apiReady) {
+			console.error('api not ready');
+			return;
+		}
+
+		let unsubscribe: () => void;
+
+		setBlocktime(api?.consts.babe?.expectedBlockTime);
```

Comment:
The way `api.consts` is used in Apps is similar. From the docs:

"Since these are constants and defined by the metadata, it is not a call, but rather the values immediately available - as you'll see in subsequent sections, there is no need for await on these, it immediately returns the type and value for you to work with."
https://polkadot.js.org/api/start/api.consts.html

Reply:
Cool, I forgot about that.